Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/Location/LocationDrawings.tsx

```diff
@@ -22,130 +22,42 @@
 //       Modified original source code; Incorporated 'Number' and 'Category' fields into 
 //       'Substation Drawings' table.
 //******************************************************************************************************
-
-
-
-
 import * as React from 'react';
 import * as _ from 'lodash';
-import { Application, OpenXDA, SystemCenter } from '@gpa-gemstone/application-typings';
-import { SystemCenter as SCGlobal } from '../global';
-import { ReactTable, Paging } from '@gpa-gemstone/react-table';
-import { useAppSelector, useAppDispatch } from '../hooks';
-import { Input, Select } from '@gpa-gemstone/react-forms';
-import { Pencil, TrashCan } from '@gpa-gemstone/gpa-symbols';
+import { SystemCenter } from '@gpa-gemstone/application-typings';
+import { GenericController, ToolTip } from '@gpa-gemstone/react-interactive';
+import LocationDrawingsTable from './LocationDrawingsTable';
+import { useAppSelector } from '../hooks';
 import { SelectRoles } from '../Store/UserSettings';
-import { ToolTip, GenericController, LoadingScreen, ServerErrorIcon } from '@gpa-gemstone/react-interactive';
-import { current } from '@reduxjs/toolkit';
+import AddEditDrawingsModal from './AddEditDrawingsModal';
 
 
-const LocationDrawingsWindow = (props: { Location: OpenXDA.Types.Location }) => {
-    const LocationDrawingController = new GenericController<SystemCenter.Types.LocationDrawing>(`${homePath}api/LocationDrawing`, "Name", true);
-    const PagingID = 'LocationDrawingPage'
-
-    const [links, setLinks] = React.useState<SystemCenter.Types.LocationDrawing[]>([]);
-    const [sortKey, setSortKey] = React.useState<keyof SystemCenter.Types.LocationDrawing>('Name');
-    const [ascending, setAscending] = React.useState<boolean>(true);
+const LocationDrawingsWindow = (props: { LocationID: number }) => {
+    const roles = useAppSelector(SelectRoles); // Deprecated
```

Comment:
If this is deprecated, can we take care of moving away from it while we are in here?

Reply:
don't know why it is marked as Deprecated. I don't think it is but I haven't double checked. @gcsantos-gpa is there a new pattern to check Roles?

Reply:
Not that I'm aware of. It seems to be something @collins-self added, maybe he could inform us with the reasoning?

Reply:
I think I misunderstood what you were saying at one point about moving to controllers over slices, so I thought his was deprecated. The comment has been removed